### PR TITLE
Support solc 0.6.12

### PR DESCRIPTION
**Pull request by desaperados**

### CODE REVIEW

**Comment by desaperados** [comment]
I think this is failing because the test suite now requires evmone. Should we add this dependency at over at nixpgs or here on dapptools?

**Comment by desaperados** [comment]
Actually - the nixpkgs solc version builds without issue. The build only fails here. And semantic tests are skipped if evmone is not present so that's not an issue.

**Comment by xwvvvvwx** [comment]
smtChecker tests are failing, maybe due to a different z3 version, they always require the latest afaik

**Comment by MrChico** [comment]
Still unclear why macos builds are failing. Could be a clang error (waiting for @leonardoalt to confirm), but I think this is fine to merge as is and if we figure out what causes the macos error we can update things

File: src/dapp/CHANGELOG.md [hunk 1]
```diff
@@ -14,12 +14,13 @@ DAPP_BUILD_EXTRACT=1 dapp test
 
 ## [0.28.0] - 2020-07-13
 ### Added
+- Support for solc 0.6.12
```
Comment (asymmetric) [inline src/dapp/CHANGELOG.md:17]: Seems like this should go under `Unreleased` :)

**Pull request merged**